src/quo/components/tags/summary_tag/view.cljs

```diff
@@ -0,0 +1,80 @@
+(ns quo.components.tags.summary-tag.view
+  (:require
+    [clojure.string :as string]
+    [quo.components.avatars.account-avatar.view :as account-avatar]
+    [quo.components.avatars.user-avatar.view :as user-avatar]
+    [quo.components.avatars.wallet-user-avatar.view :as wallet-user-avatar]
+    [quo.components.markdown.text :as text]
+    [quo.components.tags.summary-tag.style :as style]
+    [quo.foundations.colors :as colors]
+    [quo.foundations.resources :as resources]
+    [quo.theme :as quo.theme]
+    [react-native.core :as rn]))
+
+(defn- left-view
+  [{:keys [type token-props account-props collectible-props user-props saved-address-props
+           network-props]}]
+  (case type
+    :account
+    [account-avatar/view
+     {:customization-color (:customization-color account-props)
+      :size                24
+      :emoji               (:emoji account-props)
+      :type                :default}]
+    :network
+    [rn/image
+     {:source (:network-logo network-props)
+      :style  {:width  24
+               :height 24}}]
+    :saved-address
+    [wallet-user-avatar/wallet-user-avatar
+     {:full-name           (:full-name saved-address-props)
+      :size                :size-24
+      :customization-color (:customization-color saved-address-props)}]
+    :collectible
+    [rn/image
+     {:source (:image collectible-props)
+      :style  style/collectible-image}]
+    :user
+    [user-avatar/user-avatar
+     {:full-name           (:full-name user-props)
+      :size                :xs
+      :profile-picture     (:profile-picture user-props)
+      :customization-color (:customization-color user-props)}]
+    :token
+    [rn/image
+     {:source (resources/get-token (keyword (string/lower-case (:symbol token-props))))
+      :style  {:width  24
+               :height 24}}]
+    nil))
+
+(defn- view-internal
+  [{:keys [account-props user-props collectible-props network-props token-props saved-address-props
+           address type
+           theme]
+    :as   props}]
+  (let [text                (case type
```

Comment:
I think we're making the api of this component really confusing here. Why not just have a "text" param and handle this outside in application code.  although I would probably change "text" to something like "label"

Reply:
I agree too, I think this component should only receive the params it needs to work. Right now it's hard to know what keys are expected in those maps (account-props, user-props, network-props, etc) to make this work.

Also a docstring specifying the props supported for each type could work :+1: